File: Online-Health-Care-System-main/healthcare-frontend/src/app/services/appointment-service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Appointment, AppointmentRequest, ApiResponse } from '../models/user-model';

@Injectable({
  providedIn: 'root',
})
export class AppointmentService {
  private baseUrl = 'http://localhost:8080/api/appointments';

  constructor(private http: HttpClient) {}

  bookAppointment(request: AppointmentRequest): Observable<ApiResponse<Appointment>> {
    return this.http.post<ApiResponse<Appointment>>(`${this.baseUrl}`, request);
  }

  getMyAppointments(): Observable<ApiResponse<Appointment[]>> {
    return this.http.get<ApiResponse<Appointment[]>>(`${this.baseUrl}/my`);
  }

  updateStatus(id: number, status: string): Observable<ApiResponse<any>> {
    return this.http.put<ApiResponse<any>>(`${this.baseUrl}/${id}/status?status=${status}`, {});
  }

  cancelAppointment(id: number): Observable<ApiResponse<any>> {
    return this.http.delete<ApiResponse<any>>(`${this.baseUrl}/${id}`);
  }
}

File: Online-Health-Care-System-main/HealthCareSystem/src/main/java/com/healthcare/services/AppointmentService.java
package com.healthcare.services;

import com.healthcare.dto.ApiResponse;
import com.healthcare.dto.AppointmentRequest;

public interface AppointmentService {

	ApiResponse<?> bookAppointment(AppointmentRequest request, String username);
	
	ApiResponse<?> getMyAppointments(String username);
	ApiResponse<?> cancelAppointment(Long id, String username);
	ApiResponse<?> getCenterAppointments(String username);
	ApiResponse<?> updateStatus(Long id, String status, String username); // this is for the admin 
}
